tensorzero-core/src/optimization/dicl.rs

```diff
@@ -645,8 +654,10 @@ pub async fn insert_dicl_examples_with_batching(
                 let output = sample
                     .stored_output
                     .as_ref()
-                    .map(StoredOutput::to_json_string)
-                    .unwrap_or_default();
+                    .ok_or_else(|| Error::new(ErrorDetails::InternalError {
+                        message: format!("Training example is missing output after validation. {IMPOSSIBLE_ERROR_MESSAGE}")
+                    }))
+                    .and_then(|stored_output| stored_output.to_json_string().map_err(Into::into))?;
```

Comment:
let's get rid of to_json_string entirely if it's just calling serde_json::to_string